docs/cni-plugin.md

```diff
@@ -92,3 +92,87 @@ This also needs `host-local` ipam plugin to be present in one of the `PATH` dire
 ```shell
 sudo --preserve-env make test-pkg-plugin
 ```
+
+## OVS Hardware Offload
+
+This provides an option to enable OVS hardware offload with ovs-cni while using OVS data-plane
+in Mellanox ConnectX-4 Lx onwards NIC hardware (Mellanox Embedded Switch or eSwitch).
+The OVS hardware offload is using SR-IOV technology with VF representor host net-device.
+The VF representor plays the same role as TAP devices in Para-Virtual (PV) setup.
+A packet sent through the VF representor on the host arrives to the VF, and a packet sent
+through the VF is received by its representor.
+
+### Prerequisites
+
+[sriov-network-device-plugin](https://github.com/intel/sriov-network-device-plugin)
+[multus-cni](https://github.com/intel/multus-cni)
+[network-resources-injector](https://github.com/intel/network-resources-injector)
+
+### Mellanox SR-IOV Configuation
+
+To enable OVS hardware offloading, create VFs from a Mellanox PF, configure VFs in `switchdev` mode and
+set `hw-offload=true` for Open vSwitch.
+For more information [refer Mellanox doc] (https://www.mellanox.com/related-docs/prod_software/ASAP2_Hardware_Offloading_for_vSwitches_User_Manual_v4.4.pdf)
+
+### Device Plugin configuation
+
+The device plugin would create the resouce pools based on the configurations given in the `/etc/pcidp/config.json`.
+This configuration file is in json format as shown below:
+
+```json
+{
+    "resourceList": [{
+            "resourceName": "mellanox_snic0",
+            "selectors": {
+                "vendors": ["15b3"],
+                "devices": ["1018"],
+                "drivers": ["mlx5_core"],
+                "pfNames": ["snic0l","snic0r"]
+            }
+        }
+    ]
+}
+```
+
+Deploy SR-IOV network device plugin as daemonset see https://github.com/intel/sriov-network-device-plugin
+
+### Network and POD configuation
+
+After deploying `multus`, `ovs-cni` and `network-resources-injector`, Create a NetworkAttachementDefinition CRD object
+with the following config.
+
+```yaml
+apiVersion: "k8s.cni.cncf.io/v1"
+kind: NetworkAttachmentDefinition
+metadata:
+  name: ovs-trunk-net
+  annotations:
+    k8s.v1.cni.cncf.io/resourceName: intel.com/mellanox_snic0
+spec:
+  config: '{
+      "cniVersion": "0.3.1",
+      "type": "ovs",
+      "bridge": "br-snic0",
+      "trunk": "1050-1059"
+    }'
+```
+
+Now deploy a pod with the following config to attach VF into container and its representor
```

Comment:
Who is expected to create `br-snic0`? How? How is a correct VF attached to that bridge?
Can you provide an example? Is there a way to do it in a k8s-friendly way, e.g. via https://github.com/nmstate/kubernetes-nmstate ?

Reply:
@dankenigsberg: There is no fundamental difference to the current veth-pair based network attachments. An ovs-cni NAD specifies an OVS bridge representing a "physical network" hosting the VLANs. 

In a bare metal deployment this bridge would be connected to the physical network through an OVS bond for redundancy. With Mellanox SmartNICs, the OVS bridge is connected to a Linux bond of the two SmartNIC PFs, as that is the way Mellanox supports OVS HW offload with LAG through the TC flower API.

The ovs-cni assumes that these OVS bridges have been set up and that the VLANs have been provisioned on the physical networks connecting the K8s workers prior to the creation of the NADs.

Automation of such network configuration is indeed highly wanted and I am very grateful for the pointer to the kubernetes-nmstate project. We will have a look there.

Reply:
Thanks. I'd appreciate if something of this great explanation finds it place into README.md.

Reply:
Sure, i've now moved it into a new document.